commandapi-platforms/commandapi-bukkit/commandapi-bukkit-nms/commandapi-bukkit-1.15/src/main/java/dev/jorel/commandapi/nms/NMS_1_15.java

```diff
@@ -958,4 +965,14 @@ public <T> T getMinecraftServer() {
 		}
 	}
 
+	@Override
+	public String extractTranslationKey(CommandSyntaxException exception) {
+		// This method assumes the exception was generated by net.minecraft.server.v1_15_R1.CommandExceptionProvider,
+		//  which has a repeated structure that holds the translation key
+		// It might work for other exception sources ¯\_(ツ)_/¯
+		Message message = exception.getRawMessage();
+		if(!(message instanceof ChatMessage chat)) return null;
+
+		return chat.getKey();
```

Comment:
Ew. While yes, there's nothing wrong with the code you've written (it's valid and works), I dislike this pattern because it makes reading this very confusing because you'd expect `newObject` to only be declared in the `if` block scope:

```java
if (!(object instanceof Class newObject)) {
  // Return here
}

// Can now use newObject here
```

Consider using a ternary operator instead:

```suggestion
		return (message instanceof ChatMessage chat) ? chat.getKey() : null;
```

Reply:
It's doubly bad in 1.19+ :P

https://github.com/JorelAli/CommandAPI/blob/73b2d122ed05cb6eaaa107bc789200ecf2cc89ee/commandapi-platforms/commandapi-bukkit/commandapi-bukkit-nms/commandapi-bukkit-1.19-common/src/main/java/dev/jorel/commandapi/nms/NMS_1_19_Common.java#L932-L945

That does make sense though. I hadn't considered using the ternary operator here, and I think it does make the code look better.

The pattern also appears in `CommandAPIHandler#wrapArgumentType` (line 822)

https://github.com/JorelAli/CommandAPI/blob/73b2d122ed05cb6eaaa107bc789200ecf2cc89ee/commandapi-core/src/main/java/dev/jorel/commandapi/CommandAPIHandler.java#L816-L829

I'll see if the ternary operator works there too.

What do you think about this pattern in `equals` methods?

https://github.com/JorelAli/CommandAPI/blob/73b2d122ed05cb6eaaa107bc789200ecf2cc89ee/commandapi-core/src/main/java/dev/jorel/commandapi/wrappers/WrapperStringReader.java#L183-L187

https://github.com/JorelAli/CommandAPI/blob/73b2d122ed05cb6eaaa107bc789200ecf2cc89ee/commandapi-core/src/main/java/dev/jorel/commandapi/exceptions/WrapperCommandSyntaxException.java#L168-L173

Reply:
Nobody stares at `equals()` methods much, and `equals()` methods are probably never touched again once written. I don't mind this in `equals()` methods.

Reply:
Of course, using a tenary operator isn't the only option, there's also, for example:

```java
Message message = exception.getRawMessage();
if (message instanceof MutableComponent component) {
    ComponentContents contents = component.getContents();
    if (contents instanceof TranslatableContents translatableContents) {
        return translatableContents.getKey();
    } else {
        return null;
    }
} else {
    return null;
}

```
Looks less lean, but there's absolutely no chance someone could misinterpret what's going on here.